src/text/asc.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Returns the ASCII value of the first character in the string
+ * @param {string} str The string to convert to ASCII
+ * @returns The ASCII value of the first character in the string
+ */
+export const asc = (str: string): number => {
+  if (Array.from(str).length === 0) {
+    return undefined;
+  }
+  const asc = str.charCodeAt(0);
+  return asc < 128 ? asc : undefined;
```

Comment:
Same as in chr.
Is <128 limit ok ?
Also consider return "" instead undefined

Reply:
Fixed by removing the limit, now it will return any value returned by charCodeAt(0).

The reason for this check was that for Unicode character it was returning very large numbers (>50k in the emoji test case), which is not a valida ASCII value.

But it is better to return the Unicode value, and if the developer wants to limit the value to valid ASCII or extended ASCII values, it is possible to do so in GeneXus.